apps/shipments/signals.py

```diff
@@ -121,26 +121,37 @@ def trackingdata_post_save(sender, **kwargs):
                                             {"type": "tracking_data.save", "tracking_data_id": instance.id})
 
 
-@receiver(post_save_changed, sender=Shipment, fields=['device', 'state'], dispatch_uid='shipment_iot_fields_post_save')
+@receiver(post_save_changed, sender=Shipment, fields=['device', 'state', 'geofences'],
+          dispatch_uid='shipment_iot_fields_post_save')
 def shipment_iot_fields_changed(sender, instance, changed_fields, **kwargs):
     if settings.IOT_THING_INTEGRATION:
         iot_client = DeviceAWSIoTClient()
 
         shadow_update = {}
-        if Shipment.device.field in changed_fields:
-            old, new = changed_fields[Shipment.device.field]
+        device_field = Shipment._meta.get_field('device')
+        if device_field in changed_fields:
+            old, new = changed_fields[device_field]
             logging.info(f'Device ID changed from {old} to {new} for Shipment {instance.id}, updating shadow')
             if old:
-                iot_client.update_shadow(old, {'deviceId': old, 'shipmentId': '', 'shipmentState': ''})
+                iot_client.update_shadow(old, {'deviceId': old, 'shipmentId': '', 'shipmentState': '', 'geofences': ''})
             if new:
                 shadow_update['deviceId'] = new
                 shadow_update['shipmentId'] = instance.id
+                shadow_update['shipmentState'] = TransitState(instance.state).name
+                shadow_update['geofences'] = instance.geofences
 
-        if Shipment.state.field in changed_fields:
-            old, new = changed_fields[Shipment.state.field]
+        state_field = Shipment._meta.get_field('state')
+        if state_field in changed_fields:
+            old, new = changed_fields[state_field]
             logging.info(f'Shipment state changed from {old} to {new} for Shipment {instance.id}, updating shadow')
             shadow_update['shipmentState'] = TransitState(instance.state).name
 
+        geofences_field = Shipment._meta.get_field('geofences')
```

Comment:
@rudnyg just want you to take a look at this change. Pretty simple. Whenever the field is updated on a shipment, it's added to the shadow if the shipment has a device. If the `geofences` array is null or empty, it'll just send a blank string to IoT.

Reply:
So will add `geofences` which can be a list of 1 to n geofence ids. I think i just need to make sure it can handle an empty string instead of an empty array which I think it already does.

Reply:
:+1: if you want me to change it to default to an empty array that's an easy fix. I just thought you were treating empty string as null, since that is what we pass for the other shadow values when they are null.

Reply:
No change needed. It just does a ` if shadow['data'].get('geofences'): ` so doesn't matter if a list or an empty string